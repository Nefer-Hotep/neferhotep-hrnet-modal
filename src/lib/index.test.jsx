import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render, cleanup } from '@testing-library/react';
import { describe, expect, test, vi, afterEach } from 'vitest';
import userEvent from '@testing-library/user-event';
import DialogModal from './index';

describe('DialogModal', () => {
  afterEach(cleanup);

  test('DialogModal component renders correctly', () => {
    const component = renderer.create(<DialogModal openModal={true} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('The DialogModal is displayed with a message', () => {
    const component = renderer.create(
      <DialogModal openModal={true} message={'Test message !'} />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('closes DialogModal on escape key press', async () => {
    const mockSetOpenModal = vi.fn();
    render(
      <DialogModal
        openModal={true}
        setOpenModal={mockSetOpenModal}
        message={'It work !'}
      />
    );

    // The modal should be in the document initially
    expect(screen.getByText('It work !'));

    // Simulate pressing the Escape key
    await userEvent.keyboard('{Escape}');

    // Assert that setOpen was called with false, indicating the modal should close
    expect(mockSetOpenModal).toHaveBeenCalledWith(false);
  });
});
