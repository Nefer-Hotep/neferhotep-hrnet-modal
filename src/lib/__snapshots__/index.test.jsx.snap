// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogModal > DialogModal component renders correctly 1`] = `
<div
  aria-modal="true"
  className="modal"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-content"
  >
    <button
      aria-label="Close"
      className="close-btn"
      onClick={[Function]}
    >
      ×
    </button>
    <p />
  </div>
</div>
`;

exports[`DialogModal > The DialogModal is displayed with a message 1`] = `
<div
  aria-modal="true"
  className="modal"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-content"
  >
    <button
      aria-label="Close"
      className="close-btn"
      onClick={[Function]}
    >
      ×
    </button>
    <p>
      Test message !
    </p>
  </div>
</div>
`;
